src/javascript/app/common/contract_manager.js

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable class-methods-use-this */
+import { triggerContractChange } from '../hooks/events';
+
+class ContractManager {
+    constructor() {
+        if (!window.contract) {
+            window.contract = {};
+        }
+    }
+
+    set(data) {
+        if (typeof data === 'object') {
```

Comment:
Is there opportunity for `data` to be `null`? Because typeof null is also object

Reply:
since we are declaring it as {} upon construction, actually there's no possibility that it will become null